orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/ExecutionLauncher.java

```diff
@@ -64,6 +75,36 @@ public T start(T execution) throws Exception {
     return execution;
   }
 
+  protected T handleStartupFailure(T execution, Throwable failure) {
+    final String canceledBy = "system";
+    final String reason = "Failed on startup: " + failure.getMessage();
+    final ExecutionStatus status = ExecutionStatus.TERMINAL;
+    executionRepository.updateStatus(execution.getId(), status);
+    Function<Execution, Execution> reloader;
+    String executionType;
+    if (execution instanceof Pipeline) {
+      executionType = "pipeline";
+      reloader = (e) -> executionRepository.retrievePipeline(e.getId());
+    } else if (execution instanceof Orchestration) {
+      executionType = "orchestration";
+      reloader = (e) -> executionRepository.retrieveOrchestration(e.getId());
+    } else {
+      log.error("Unknown execution type: " + execution.getClass().getSimpleName());
+      executionType = "unknown";
+      reloader = (e) -> {
+        e.setCancellationReason(reason);
+        e.setCanceled(true);
+        e.setCanceledBy(canceledBy);
+        e.setStatus(status);
+        return e;
+      };
```

Comment:
maybe overcomplifying things, but I felt like throwing an "I dont know what type of Execution this is" exception here would be worse than just munging the object and handing it back since this path was about handling an error in a place we don't have a good story around handling errors in the first place